packages/gatsby-source-wordpress/src/steps/source-nodes/create-nodes/create-remote-media-item-node.js

```diff
@@ -255,110 +256,138 @@ export const createRemoteMediaItemNode = async ({
-
-      const auth =
-        htaccessCredentials && shouldUseHtaccessCredentials
-          ? {
-              htaccess_pass: htaccessCredentials?.password,
-              htaccess_user: htaccessCredentials?.username,
-            }
-          : null
-
-      // if this errors, it's caught one level above in fetch-referenced-media-items.js so it can be placed on the end of the request queue
-      const node = await createRemoteFileNode({
-        url: mediaItemUrl,
-        auth,
-        ...createFileNodeRequirements,
-        pluginOptions,
-      })
+    if (!imageNode) {
+      imageNode = await retry(
+        async () => {
+          if (fetchState.shouldBail) {
+            failedImageUrls.add(mediaItemUrl)
+            return null
+          }
 
-      return node
-    },
-    {
-      retries: 3,
-      factor: 1.1,
-      minTimeout: 5000,
-      onRetry: error =>
-        errorPanicker({
-          error,
-          reporter,
-          node: mediaItemNode,
-          fetchState,
-          parentName,
-        }),
+          const { hostname: wpUrlHostname } = url.parse(wpUrl)
+          const { hostname: mediaItemHostname } = url.parse(mediaItemUrl)
+
+          const htaccessCredentials = pluginOptions.auth.htaccess
+
+          // if media items are hosted on another url like s3,
+          // using the htaccess creds will throw 400 errors
+          const shouldUseHtaccessCredentials =
+            wpUrlHostname === mediaItemHostname
+
+          const auth =
+            htaccessCredentials && shouldUseHtaccessCredentials
+              ? {
+                  htaccess_pass: htaccessCredentials?.password,
+                  htaccess_user: htaccessCredentials?.username,
+                }
+              : null
+
+          // if this errors, it's caught one level above in fetch-referenced-media-items.js so it can be placed on the end of the request queue
+          const node = await createRemoteFileNode({
+            url: mediaItemUrl,
+            auth,
+            ...createFileNodeRequirements,
+            pluginOptions,
+          })
+
+          return node
+        },
+        {
+          retries: 3,
+          factor: 1.1,
+          minTimeout: 5000,
+          onRetry: error =>
+            errorPanicker({
+              error,
+              reporter,
+              node: mediaItemNode,
+              fetchState,
+              parentName,
+            }),
+        }
+      )
     }
-  )
+  }
 
-  if (!remoteFileNode) {
+  if (!imageNode) {
     return null
   }
 
   // push it's id and url to our store for caching,
   // so we can touch this node next time
   // and so we can easily access the id by source url later
   store.dispatch.imageNodes.pushNodeMeta({
-    id: remoteFileNode.id,
+    id: imageNode.id,
     sourceUrl: mediaItemUrl,
     modifiedGmt,
   })
 
-  if (hardCacheMediaFiles) {
+  if (hardCacheMediaFiles && shouldFetchImage) {
```

Comment:
shouldFetchImage is true when the fileSize is greater than the max. Reading this if statement makes sense, but it seems like the `shouldFetchImage` part of this boolean should be inverted. Right now it evaluated to this... which seems incorrect

```
if (hardCacheMediaFiles && (fileSize > maxFileSizeBytes)) {

}
```

Reply:
Good catch! My terrible variable naming came back to bite here :grimacing: